### ISSUE TITLE
Sentiment task: finding best hyperparameters.

### ISSUE THREAD

**Issue by diogo-cruz** [EdoardoPona/predicting-inductive-biases-RL#17]
With:
n_envs: 4
n_steps: 64
batch_size: 64
learning_rate: 0.00001
n_epochs: 5
ent_coef: 0.0
I can bring coeff to 0.01, and target_kl to 0.01 without any visible performance loss (when compared to coeff=0.0). Decreasing n_steps leads to a noticeable penalty, while increasing the batch size improves things (but can't really be done without reducing n_envs, since there is no more RAM).

Note that n_envs = 4 is the limit of what the instance can take, so if you plan to have more than one person using the same instance, it's better to reduce that number.

**Comment by diogo-cruz** [comment]
We should probably also play around with the entropy coefficient, since the model is currently mode collapsing hard.

**Comment by diogo-cruz** [comment]
Setup that maxes out `gpt2` model in g5.xlarge instance:
```
prompt_length = 16
episode_length = 48

n_envs: 4

alg:
  id: ppo
  args:
    n_steps: 128
    batch_size: 64
    verbose: 0
    learning_rate: 0.000001
    n_epochs: 5
    device: cuda
  kl_div:
    coeff: 0.1
    target_kl: 0.2
```

Setup that maxes out `gpt2-medium` model in g5.xlarge instance:
```
prompt_length = 16
episode_length = 16

n_envs: 1

alg:
  id: ppo
  args:
    n_steps: 128
    batch_size: 32 # reduced due to lack of VRAM, originally 64
    verbose: 0
    learning_rate: 0.000001
    n_epochs: 5
    device: cuda
  kl_div:
    coeff: 0.01 # 0.1 is also ok
    target_kl: 0.2
```

Note that, if using these settings, the `imdb_dataset_preprocessing.py` script should be run with `--max_tokens=16`.
